BeeSwift/GoalViewController.swift

```diff
@@ -284,7 +287,70 @@ class GoalViewController: UIViewController,  UIScrollViewDelegate, DatapointTabl
         setValueTextField()
         updateInterfaceToMatchGoal()
     }
-
+    
+    enum MenuAction {
+        case goalCommitment
+        case goalStop
+        case goalData
+        case goalStatistics
+        case goalSettings
+        
+        func makeLink(username: String, goalName: String) -> URL? {
+            guard
+                let accessToken = ServiceLocator.currentUserManager.accessToken
+            else { return nil }
+            
+            let destinationUrl: URL
+            
+            switch self {
+            case .goalCommitment:
+                destinationUrl = DeeplinkGenerator.generateDeepLinkToGoalCommitment(username: username, goalName: goalName)
+            case .goalStop:
+                destinationUrl = DeeplinkGenerator.generateDeepLinkToGoalStop(username: username, goalName: goalName)
+            case .goalData:
+                destinationUrl = DeeplinkGenerator.generateDeepLinkToGoalData(username: username, goalName: goalName)
+            case .goalStatistics:
+                destinationUrl = DeeplinkGenerator.generateDeepLinkToGoalStatistics(username: username, goalName: goalName)
+            case .goalSettings:
+                destinationUrl = DeeplinkGenerator.generateDeepLinkToGoalSettings(username: username, goalName: goalName)
+            }
+            
+            return DeeplinkGenerator.generateDeepLinkToUrl(accessToken: accessToken, username: username, url: destinationUrl)
+        }
+    }
+    
+    struct MenuOption {
+        let title: String
+        let action: MenuAction
+        let imageSystemName: String
+    }
+    
+    private func getMenuOptions() -> [MenuOption] {
+        [
+            MenuOption(title: "Commitment", action: .goalCommitment, imageSystemName: "signature"),
+            MenuOption(title: "Stop/Pause", action: .goalStop, imageSystemName: "pause.fill"),
+            MenuOption(title: "Data", action: .goalData, imageSystemName: "tablecells"),
+            MenuOption(title: "Statistics", action: .goalStatistics, imageSystemName: "chart.bar.fill"),
+            MenuOption(title: "Settings", action: .goalSettings, imageSystemName: "gearshape.2"),
+        ]
+    }
+    
+    private func createGoalMenu() -> UIMenu {
+        let options = getMenuOptions()
+        let actions = options.map { option in
+            UIAction(title: option.title, image: UIImage(systemName: option.imageSystemName), handler: { [weak self] _ in
+                guard let self else { return }
```

Comment:
This is a fun little bit of syntax. 👍

Reply:
ah it was very common for a while to declare weak self then avoid self? it each time by only executing the block when self was then non-nil

And instead of `guard let self = self else { return }` we may have seen `guard let strongSelf = self else...`.
Thanks to the syntax highlighting in Xcode we can usually see which one is which (local, strong self, vs local, weak self vs the entity's own self). And of course the compiler calculates it for us too.